octree/basic.go

```diff
@@ -131,8 +131,13 @@ func (octree *basicOctree) At(x, y, z float64) (pc.Data, bool) {
 	return nil, false
 }
 
-// Iterate TODO: Implement iterate for octree.
+// Iterate takes in the number of batches, a batch id and a function to apply to each point in a basic octree
+// until false is returned. These parameters are passing them to iterateRecursive along with the starting index
+// (0) to iterate though a basic octree if the given number of batches is greater than 0.
 func (octree *basicOctree) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d pc.Data) bool) {
+	if numBatches >= 0 {
+		octree.iterateRecursive(numBatches, myBatch, 0, fn)
```

Comment:
I'd prefer to keep it as is as iterateRecursive implies to me that that this is purely a helper for iterate while iterateRecursively sounds more like a fully-separate type of iterate function that is being done recursively  as oppose to the non-recursive one.

Reply:
would we want the name to be something like `helperIterateRecursive` so that the name is different from `iterate`? I am fine if we want to leave it as is too